### PR TITLE
feat: Conditional redirection to the onboarding guide

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes enable conditional redirection to the onboarding guide & some UI/UX changes around the same. Here're the conditions,
**1. On the first login,** 
- If helm charts detected (apart from devtron-operator) > Show helm apps list
- If no helm charts are detected apart from the devtron-operator > Redirect to the onboarding view

**Without CI/CD installed**
Screenshot 2022-12-22 at 6 58 30 PM

**With CI/CD installed**
Screenshot 2022-12-22 at 6 58 38 PM

**2. Normal login,**
- If helm charts detected (apart from devtron-operator) > Show helm apps list
- If no helm charts are detected apart from the devtron-operator > Show the list along with guided content cards

**Without CI/CD installed**
Screenshot 2022-12-22 at 6 57 49 PM

**With CI/CD installed**
Screenshot 2022-12-22 at 6 58 13 PM

Fixes - [AB#1248](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1248), [AB#1398](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1398)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually in both with & without CI/CD installed modes by,

1. Fresh installation without CI/CD & logging in for the first time > redirected to guided onboarding page
2. Fresh installation without CI/CD & logging in normally,
- Redirected to helm app listing > loaded app listing view > devtron-operator app is loaded along with the guided content cards
- New helm app created > loaded app listing view > helm apps are loaded but no guided content cards are displayed
4. Installing CI/CD integration & trying to log in again 
- Redirected to devtron app listing > loaded app listing view > listing view loaded without any apps & with the guided content cards
- New devtron app created > loaded app listing view > devtron app loaded but no guided content cards are displayed

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/list-new/HelmAppList.tsx [hunk 1]
```diff
@@ -357,7 +370,68 @@ export default function HelmAppList({
...
+            <div className="bcn-0" key={index}>
+                <div className="h-8"></div>
+                <div className="ea-fetch-error-message above-header-message flex left">
+                    <span className="mr-8 flex">
+                        <ErrorExclamationIcon className="icon-dim-20" />
+                    </span>
+                    <span>{externalHelmListFetchError}</span>
+                    <CloseIcon
+                        className="icon-dim-24 dc__align-right cursor"
+                        onClick={() => _removeExternalAppFetchError(index)}
```
Comment (vivek-devtron) [inline src/components/app/list-new/HelmAppList.tsx:387]: Extract this inline function

File: src/components/app/list-new/HelmAppList.tsx [hunk 2]
```diff
@@ -372,79 +446,47 @@ export default function HelmAppList({
...
+                {filteredHelmAppsList
+                    .slice(payloadParsedFromUrl.hOffset, payloadParsedFromUrl.hOffset + payloadParsedFromUrl.size)
+                    .map((app) => renderHelmAppLink(app))}
+                {showGuidedContentCards && (
+                    <div className="helm-app-guided-cards-wrapper">
+                        <ContentCard
+                            redirectTo={URLS.GLOBAL_CONFIG_CLUSTER}
+                            direction={CardContentDirection.Horizontal}
+                            imgSrc={HelmCluster}
+                            title="Discover and manage existing helm releases via GUI"
```
Comment (vivek-devtron) [inline src/components/app/list-new/HelmAppList.tsx:470]: move messaging to const

File: src/components/common/ContentCard/ContentCard.scss [hunk 3]
```diff
@@ -0,0 +1,30 @@
+.content-card-container {
+    &:hover {
+        box-shadow: 0 8px 12px 0 rgba(30, 35, 96, 0.1), 0 1px 4px 0 rgba(0, 0, 0, 0.1);
+        border-radius: 400;
+    }
+
+    .Horizontal {
```
Comment (vivek-devtron) [inline src/components/common/ContentCard/ContentCard.scss:7]: Why are we using capitalize name?

File: src/components/onboardingGuide/OnboardingGuide.tsx [hunk 4]
```diff
@@ -67,64 +62,43 @@ function OnboardingGuide({
...
-                                <div className="fw-6 fs-16 pt-32 pb-32 pl-24 pr-24">
-                                    Deploy custom applications using CI/CD pipelines
-                                </div>
-                            </NavLink>
-                        </div>
+                        <ContentCard
+                            redirectTo={URLS.CHARTS_DISCOVER}
+                            onClick={onClickHelmChart}
+                            imgSrc={HelmCollage}
+                            title="Deploy and manage your favourite Kubernetes packages"
```
Comment (vivek-devtron) [inline src/components/onboardingGuide/OnboardingGuide.tsx:74]: Move messaging to const

File: src/components/app/list-new/HelmAppList.tsx [hunk 5]
```diff
@@ -254,80 +283,76 @@ export default function HelmAppList({
...
         return (
             <div className="app-list__header">
                 <div className="app-list__cell--icon"></div>
                 <div className="app-list__cell app-list__cell--name">
-                    {sseConnection && <span>App/Release name</span>}
+                    {sseConnection && <span>{HELM_APP_LIST_HEADERS.AppName}</span>}
                     {!sseConnection && (
                         <button
                             className="app-list__cell-header"
                             onClick={(e) => {
```
Comment (vivek-devtron) [inline src/components/app/list-new/HelmAppList.tsx:350]: extract this inline method

File: src/components/app/list/AppListView.tsx [hunk 6]
```diff
@@ -108,43 +115,73 @@ export class AppListView extends Component<AppListViewProps>{
...
+                    <div className="dc__loading-wrapper">
+                        <Progressing pageLoader />
+                    </div>
+                </React.Fragment>
+            )
+        } else if (this.props.view === AppListViewType.EMPTY) {
+            return this.renderGuidedCards()
+        } else if (this.props.view === AppListViewType.NO_RESULT) {
+            return (
+                <React.Fragment>
```
Comment (vivek-devtron) [inline src/components/app/list/AppListView.tsx:160]: Can we remove this `<React.Fragment>`?

File: src/components/app/list/AppListView.tsx [hunk 7]
```diff
@@ -108,43 +115,73 @@ export class AppListView extends Component<AppListViewProps>{
...
+                        title={'No apps found'}
+                        message={"We couldn't find any matching applications."}
+                        buttonLabel={'Clear filters'}
+                        clickHandler={this.props.clearAll}
+                    />
+                </React.Fragment>
+            )
+        } else if (this.props.view === AppListViewType.ERROR) {
+            return (
+                <React.Fragment>
```
Comment (vivek-devtron) [inline src/components/app/list/AppListView.tsx:172]: same here

File: src/components/app/list/AppListView.tsx [hunk 8]
```diff
@@ -108,43 +115,73 @@ export class AppListView extends Component<AppListViewProps>{
...
-                </div>
-            </React.Fragment>
-        }
-        else {
-            return <React.Fragment>
-                {this.renderAppList()}
-                {this.renderPagination()}
-            </React.Fragment>
+            return (
+                <React.Fragment>
```
Comment (vivek-devtron) [inline src/components/app/list/AppListView.tsx:150]: Can we remove this <React.Fragment>?

File: src/components/common/ContentCard/ContentCard.types.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+export enum CardContentDirection {
```
Comment (vivek-devtron) [inline src/components/common/ContentCard/ContentCard.types.ts:1]: please format this fime

File: src/components/app/list-new/HelmAppList.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/app/list-new/HelmAppList.tsx]: Done with the changes. Using dataset attr for getting index value.

File: src/components/app/list-new/HelmAppList.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/app/list-new/HelmAppList.tsx]: Done

File: src/components/common/ContentCard/ContentCard.scss (hunk 3 above)
Comment (sohel-rp) [reply src/components/common/ContentCard/ContentCard.scss]: I was using an enum value directly. Changed it to small case.

File: src/components/onboardingGuide/OnboardingGuide.tsx (hunk 4 above)
Comment (sohel-rp) [reply src/components/onboardingGuide/OnboardingGuide.tsx]: Done

File: src/components/app/list-new/HelmAppList.tsx (hunk 5 above)
Comment (sohel-rp) [reply src/components/app/list-new/HelmAppList.tsx]: Done

File: src/components/app/list/AppListView.tsx (hunk 6 above)
Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx]: Done

File: src/components/app/list/AppListView.tsx (hunk 7 above)
Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx]: Done

File: src/components/common/ContentCard/ContentCard.types.ts (hunk 9 above)
Comment (sohel-rp) [reply src/components/common/ContentCard/ContentCard.types.ts:1]: Done

File: src/components/app/list/AppListView.tsx (hunk 8 above)
Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx]: Done

**Review by vivek-devtron — APPROVED** [review]

File: src/components/onboardingGuide/OnboardingGuide.tsx [hunk 10]
```diff
@@ -65,66 +66,47 @@ function OnboardingGuide({
...
-                                    className="onboarding-card__img dc__top-radius-4"
-                                    src={DeployCICD}
-                                    alt="Please connect cluster"
-                                />
-                                <div className="fw-6 fs-16 pt-32 pb-32 pl-24 pr-24">
-                                    Deploy custom applications using CI/CD pipelines
-                                </div>
-                            </NavLink>
-                        </div>
+                        <ContentCard
```
Comment (shivani170) [inline src/components/onboardingGuide/OnboardingGuide.tsx:76]: Could` we rename it to more specific like `OnboardingContentCard`

File: src/components/app/list/AppListView.tsx [hunk 11]
```diff
@@ -108,43 +115,67 @@ export class AppListView extends Component<AppListViewProps>{
...
-                    title={"No apps found"}
+            )
+        } else if (this.props.view === AppListViewType.EMPTY) {
+            return this.renderGuidedCards()
+        } else if (this.props.view === AppListViewType.NO_RESULT) {
+            return (
+                <Empty
+                    view={this.props.view}
+                    title={'No apps found'}
                     message={"We couldn't find any matching applications."}
```
Comment (shivani170) [inline src/components/app/list/AppListView.tsx:161]: `message={"We couldn't find any matching applications."}` Can Extract this messaging in constant as well

File: src/components/app/list/AppListView.tsx [hunk 12]
```diff
@@ -108,43 +115,67 @@ export class AppListView extends Component<AppListViewProps>{
...
-        }
-        else {
-            return <React.Fragment>
-                {this.renderAppList()}
-                {this.renderPagination()}
-            </React.Fragment>
+            )
+        } else {
+            return (
+                <React.Fragment>
```
Comment (shivani170) [inline src/components/app/list/AppListView.tsx:174]: Please remove React.Fragment

File: src/components/onboardingGuide/OnboardingGuide.tsx (hunk 10 above)
Comment (sohel-rp) [reply src/components/onboardingGuide/OnboardingGuide.tsx:76]: @shivani170 It's not specific to Onboarding. This component is common & can be used anywhere which needs content to be shown in the same design.

Comment (sohel-rp) [reply src/components/onboardingGuide/OnboardingGuide.tsx:76]: Done

File: src/components/app/list/AppListView.tsx (hunk 11 above)
Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx]: Done

Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx]: Also, moved some other consts & made some related changes.

File: src/components/app/list/AppListView.tsx (hunk 12 above)
Comment (sohel-rp) [reply src/components/app/list/AppListView.tsx:180]: Done

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**